core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionAttributes.java

```diff
@@ -48,11 +66,61 @@
         return this;
     }
 
+    /**
+     * Merge attributes of a higher precedence into the current lower precedence collection.
+     */
+    public void merge(ExecutionAttributes higherPrecedenceExecutionAttributes) {
+        higherPrecedenceExecutionAttributes.getAttributes().forEach(this.attributes::put);
+    }
+
```

Comment:
This is opposite of what `AttributeMap` does; they should be identical so that `this` has higher precedence. Then in `AwsClientHandlerUtils` we can use the attributes on the request as the 'base', and merge the lower precedence attributes into it.